src/Command/SingleDirectoryComponent.php

```diff
@@ -0,0 +1,218 @@
+<?php declare(strict_types = 1);
+
+namespace DrupalCodeGenerator\Command;
+
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Extension\ThemeHandlerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Asset\LibraryDiscovery;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Extension\ExtensionLifecycle;
+use DrupalCodeGenerator\Application;
+use DrupalCodeGenerator\Asset\AssetCollection;
+use DrupalCodeGenerator\Asset\File;
+use DrupalCodeGenerator\Attribute\Generator;
+use DrupalCodeGenerator\GeneratorType;
+use DrupalCodeGenerator\InputOutput\Interviewer;
+use DrupalCodeGenerator\Utils;
+use DrupalCodeGenerator\Validator\Choice;
+use DrupalCodeGenerator\Validator\Optional;
+use DrupalCodeGenerator\Validator\Required;
+use DrupalCodeGenerator\Validator\RequiredMachineName;
+use Symfony\Component\Console\Question\Question;
+
+#[Generator(
+  name: 'single-directory-component',
+  description: 'Generates Drupal SDC theme component',
+  aliases: ['sdc'],
+  templatePath: Application::TEMPLATE_PATH . '/_sdc',
+  type: GeneratorType::THEME,
+)]
+final class SingleDirectoryComponent extends BaseGenerator implements ContainerInjectionInterface {
+  private const COMPONENT_PATH_TOKEN = '{machine_name}' . \DIRECTORY_SEPARATOR . '{directory}' . \DIRECTORY_SEPARATOR . '{component_machine_name}' . \DIRECTORY_SEPARATOR;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    private readonly ModuleHandlerInterface $moduleHandler,
+    private readonly ThemeHandlerInterface $themeHandler,
+    private readonly LibraryDiscovery $libraryDiscovery,
+  ) {
+    parent::__construct();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): self {
+    return new self(
+      $container->get('module_handler'),
+      $container->get('theme_handler'),
+      $container->get('library.discovery'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function generate(array &$vars, AssetCollection $assets): void {
+    $this->askQuestions($vars);
+    $this->generateAssets($vars, $assets);
+  }
+
+  private function askQuestions(array &$vars): array {
+    $ir = $this->createInterviewer($vars);
+    $vars['name'] = $ir->askName();
+    $vars['machine_name'] = $ir->askMachineName();
```

Comment:
Ask machine name first. So the IR will figure out the theme human name without interaction.

Reply:
When asking the human name first, IR is able to infer the machine name as well.
But I agree with you, it makes more sense to ask for an id first.

In another PR, I could also add support for autocomplete. If the theme entered doesn't already exists, maybe trigger the theme generator?

Reply:
> IR is able to infer the machine name as well.
Right, it'll set it as default value. But it'll still ask for machine name as theoretically there might more than one theme with the same human name.

Reply:
> If the theme entered doesn't already exists, maybe trigger the theme generator?

I cannot imagine a case when someone wants to generate a component before generating a theme. Triggering theme generator will make this generator too complicated. For module component generators we simply create an info file (#88). I think it makes sense to do the same for theme component generators.